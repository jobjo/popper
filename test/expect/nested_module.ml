open Popper
open Sample.Syntax

module X = struct
  module Y = struct
    module Z = struct
      type t = int [@@deriving show, popper, ord]
      type foo = int [@@deriving show, popper, ord]
    end
  end
end

type k =
  { t : X.Y.Z.t
  ; foo : X.Y.Z.foo
  }
[@@deriving show, popper, ord]

let test =
  test @@ fun () ->
  let* x = k_sample in
  equal k_comparator x x

let () = Clean.run test
