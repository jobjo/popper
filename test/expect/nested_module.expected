
1/1 tests passed in XXs:

  âœ“  Anonymous  Passed 300 samples

Test Successful: 1/1 tests passed in XXs.

